taichi/analysis/value_diff.cpp

```diff
@@ -111,6 +154,21 @@ DiffRange value_diff_loop_index(Stmt *stmt, Stmt *loop, int index_id) {
   return diff.run();
 }
 
+int value_diff_ptr_index(Stmt *val1, Stmt *val2) {
+  // Return the *absolute* difference of the value of two statements.
+  // Return -1 if the relationship is uncertain.
```

Comment:
Returning the absolute value here is a bit confusing - maybe we can return a `std::pair<bool, int>`, where the `bool` is `true` for `certain` and `false` for uncertain, and the `int` is for the difference (without taking abs)?